teacherchannel/database.py

```diff
@@ -0,0 +1,151 @@
+                    ),
+                    TeacherChannel.guild_id == guild_id,
+                )
+            )
+            .one_or_none()
+        )
+
+    @staticmethod
+    def get_all(guild_id: int) -> List[TeacherChannel]:
+        return session.query(TeacherChannel).filter_by(guild_id=guild_id).all()
+
+    @staticmethod
+    def get_guilds() -> List[TeacherChannel]:
+        return (
+            session.query(TeacherChannel.guild_id)
+            .distinct(TeacherChannel.guild_id)
+            .all()
+        )
+
+    @staticmethod
+    def add_channel(
+        guild_id: int, master_id: int, slave_id: int
+    ) -> Optional[TeacherChannel]:
+        channel = TeacherChannel.get(guild_id, master_id)
+        if channel is not None:
+            return None
+        channel = TeacherChannel(
+            guild_id=guild_id, master_id=master_id, slave_id=slave_id
+        )
+        session.add(channel)
+        session.commit()
+        return channel
+
+    @staticmethod
+    def add_teacher(
+        guild_id: int, channel_id: int, teacher_id: int
+    ) -> Optional[TeacherChannel]:
+        channel = TeacherChannel.get(guild_id, channel_id)
+        if not channel:
+            return None
+        if teacher_id in [i.user_id for i in channel.teachers]:
+            return None
+        channel.teachers.append(Teacher(user_id=teacher_id))
+        session.merge(channel)
+        session.commit()
+        return channel
+
+    @staticmethod
+    def remove_channel(guild_id: int, channel_id) -> bool:
+        channel = TeacherChannel.get(guild_id, channel_id)
+        if not channel:
+            return False
+        session.delete(channel)
+        session.commit()
+        return True
+
+    @staticmethod
+    def remove_teacher(guild_id, channel_id, teacher_id) -> bool:
+        channel = TeacherChannel.get(guild_id, channel_id)
+        if not channel:
+            return False
+        teacher_to_remove = Teacher.get(channel.slave_id, teacher_id)
+        print(teacher_to_remove)
+        try:
+            channel.teachers.remove(teacher_to_remove)
+        except ValueError:
+            return False
+        session.merge(channel)
+        session.commit()
+        return True
+
+    def __repr__(self) -> str:
+        return (
+            f"<TeacherChannel guild_id='{self.guild_id}' "
+            f"slave_id='{self.slave_id}' master_id='{self.master_id}' "
+            f"teachers=[{'; '.join([str(i.user_id) for i in self.teachers])}]>"
+        )
+
+    def dump(self) -> dict:
+        return {
+            "guild_id": self.guild_id,
+            "slave_id": self.slave_id,
+            "master_id": self.master_id,
+            "teachers": [i.user_id for i in self.teachers],
+        }
+
+
+class Teacher(database.base):
+    """Teachers of a channel"""
+
+    __tablename__ = "school_teacherchannel_teacher"
+
+    id = Column(BigInteger, primary_key=True, autoincrement=True)
```

Comment:
Usually in Pumpkin databases ID variable is named IDX and it's only Integer (not BigInteger).

Also, do we need this ID? Don't we have any unique value combination, such as slave_id and user_id?